.github/workflows/flow-pull-request-checks.yaml

```
@@ -0,0 +1,39 @@
+name: PR Checks
+on:
+  workflow_dispatch:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+concurrency:
+  group: pr-checks-${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  test:
+    name: Tests
+    runs-on: [self-hosted, Linux, medium, ephemeral]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install packages
+        run: npm ci
```

We need to add a `setup-node` task because not all runners are guaranteed to have the correct version of NodeJS preinstalled. 

```suggestion
      - name: Install NodeJS
         uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c
         with:
            node-version: '16'
      
      - name: Install packages
        run: npm ci
```